pkg/db/postgres.go

```diff
@@ -160,3 +164,412 @@ func (m *Postgres) PrintEnvironmentTestsAndTestCases(w http.ResponseWriter, _ *h
 	// Close the HTML page structure
 	fmt.Fprintf(w, "</body></html>")
 }
+
+// ServeTestCharts writes the individual test charts to a JSON HTTP response
+func (m *Postgres) ServeTestCharts(w http.ResponseWriter, r *http.Request) {
+	start := time.Now()
+	queryValues := r.URL.Query()
+	env := queryValues.Get("env")
+	if env == "" {
+		http.Error(w, "missing environment name", http.StatusInternalServerError)
+		return
+	}
+	test := queryValues.Get("test")
+	if test == "" {
+		http.Error(w, "missing test name", http.StatusInternalServerError)
+		return
+	}
+
+	var validEnvs []string
+	err := m.db.Select(&validEnvs, "SELECT DISTINCT EnvName FROM db_environment_tests")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to execute SQL query for list of valid environments: %v", err), http.StatusInternalServerError)
```

Comment:
```suggestion
		http.Error(w, fmt.Sprintf("failed to execute SQL query for list of valid environments: %v", err), http.StatusUnprocessableEntity)
```

Throwing a 500 doesn't make sense here as that means there's something wrong with the server, but the problem is with the input not having an expected value. I'd recommend using a 422 instead.